src/main/java/com/rocketden/main/service/SubmitService.java

```diff
@@ -0,0 +1,16 @@
+package com.rocketden.main.service;
+
+import com.rocketden.main.dao.RoomRepository;
+import org.springframework.stereotype.Service;
+
+/**
+ * Class to handle code updates and miscellaneous requests.
+ */
+@Service
+public abstract class SubmitService extends GameManagementService {
+
+    protected SubmitService(RoomRepository repository, SocketService socketService) {
+        super(repository, socketService);
+    }
```

Comment:
Same as above 

Also, something that might be worth us looking into is [inheritance vs. composition](https://stackoverflow.com/questions/2399544/difference-between-inheritance-and-composition) since apparently sometimes inheritance isn't the best way to go

Reply:
Good discussion about this -- I'm changing this to composition across the PR.